src/Polly.Core/Simmy/Outcomes/ChaosOutcomeStrategy.cs

```diff
@@ -13,7 +13,7 @@ public ChaosOutcomeStrategy(ChaosOutcomeStrategyOptions<T> options, ResilienceSt
     {
         _telemetry = telemetry;
         _onOutcomeInjected = options.OnOutcomeInjected;
-        _outcomeGenerator = options.OutcomeGenerator;
+        _outcomeGenerator = options.OutcomeGenerator!;
```

Comment:
If this is nullable now, why isn't `_outcomeGenerator` changed to allow nullable and the appropriate null guards made in `ExecuteCore`?

Reply:
Because it won't ever be null, if `OutcomeGenerator` is null, the validation fails